File: prototypes/angular-node-javascript/backend-node-javascript/src/features/person/person-controller.js
class PersonController {
  constructor(service) {
    this.service = service;
  }

  getItems = async (req, res) => {
    try {
      const items = await this.service.getItems();
      res.json(items);
    } catch (error) {
      res.status(500).json({ error: error.message });
    }
  };
}

export default PersonController;

File: backend-java/03-backend-java/src/main/java/com/ganatan/modules/person/PersonController.java
package com.ganatan.modules.person;

import java.io.IOException;
import java.util.List;
import jakarta.servlet.ServletException;
import jakarta.servlet.annotation.WebServlet;
import jakarta.servlet.http.HttpServlet;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import com.fasterxml.jackson.databind.ObjectMapper;


import com.ganatan.config.AppConfig;

@WebServlet("/persons")
public class PersonController extends HttpServlet {

    private final PersonService service;

    public PersonController() {
        PersonRepository repository = new PersonRepository(AppConfig.useDatabase());
        this.service= new PersonService(repository);
    }

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
    	List<Person> items = service.getItems();

        response.setContentType("application/json");
        response.setCharacterEncoding("UTF-8");
        new ObjectMapper().writeValue(response.getWriter(), items);
    }
}
